code/modules/karma/karma.dm

```diff
@@ -186,418 +188,17 @@ GLOBAL_LIST_EMPTY(karma_spenders)
 		to_chat(src, "<span class='warning'>Karma is disabled.</span>")
 		return
 
-	var/currentkarma = verify_karma()
-	if(!isnull(currentkarma))
-		to_chat(usr, {"<br>You have <b>[currentkarma]</b> available."})
-
-/client/proc/verify_karma()
-	var/currentkarma = 0
-	if(!SSdbcore.IsConnected())
-		to_chat(usr, "<span class='warning'>Unable to connect to karma database. Please try again later.<br></span>")
-		return
-
-	var/datum/db_query/query = SSdbcore.NewQuery("SELECT karma, karmaspent FROM karmatotals WHERE byondkey=:ckey", list(
-		"ckey" = ckey
-	))
-	if(!query.warn_execute())
-		qdel(query)
-		return
-
-	var/totalkarma
-	var/karmaspent
-	while(query.NextRow())
-		totalkarma = query.item[1]
-		karmaspent = query.item[2]
-	qdel(query)
-	currentkarma = (text2num(totalkarma) - text2num(karmaspent))
-
-	return currentkarma
+	var/available_karma = karmaholder.karma_earned - karmaholder.karma_spent
+	to_chat(usr, "You have ["<b>[available_karma]</b>" || "a database error. Contact the host if they are"] available.")
 
 /client/verb/karmashop()
 	set name = "karmashop"
-	set desc = "Spend your hard-earned karma here"
```

Comment:
This is the most important part of this PR

Reply:
"Hard earned"